apps/web/src/components/attachment/index.tsx

```diff
@@ -272,6 +273,7 @@ const AttachmentMenuItems: (
                   onClick: () => {
                     useEditorStore.getState().openSession(note.id);
                     DialogManager.closeAll();
+                    hashNavigate("/", { replace: true, notify: false });
```

Comment:
This feels like a temporary solution. Why exactly can't the settings open after navigation and why do we need this?

Reply:
When the settings dialog is closed, we reset the hash router:
https://github.com/streetwriters/notesnook/blob/master/apps/web/src/navigation/hash-routes.tsx#L77

But in this attachment component, we call the `closeAll` method, which doesn't trigger the close action for the dialog, just removes it from the DOM

That's why I'm resetting the hash router manually here